src/transformers/models/molmo/modular_molmo.py

```diff
@@ -0,0 +1,2289 @@
+            Number of hidden layers in the Transformer encoder.
+        num_attention_heads (`int`, *optional*, defaults to 12):
+            Number of attention heads for each attention layer in the Transformer encoder.
+        image_size (`int`, *optional*, defaults to 224):
+            The size (resolution) of each image.
+        patch_size (`int`, *optional*, defaults to 32):
+            The size (resolution) of each patch.
+        hidden_act (`str` or `function`, *optional*, defaults to `"quick_gelu"`):
+            The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
+            `"relu"`, `"selu"` and `"gelu_new"` `"quick_gelu"` are supported.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the layer normalization layers.
+        attention_dropout (`float`, *optional*, defaults to 0.0):
+            The dropout ratio for the attention probabilities.
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+    Example:
+
+    ```python
+    >>> from transformers import MolmoVisionConfig, MolmoVisionModel
+
+    >>> # Initializing a MolmoVisionConfig with allenai/Molmo-7B-D-0924-hf style configuration
+    >>> configuration = MolmoVisionConfig()
+
+    >>> # Initializing a MolmoVisionModel (with random weights) from the allenai/Molmo-7B-D-0924-hf style configuration
+    >>> model = MolmoVisionModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    def __init__(
+        self,
+        hidden_size=1024,
+        num_attention_heads=16,
+        intermediate_size=4096,
+        num_hidden_layers=23,
+        num_image_positions=577,
+        image_size=576,
+        patch_size=14,
+        hidden_act="quick_gelu",
+        layer_norm_eps=1e-5,
+        attention_dropout=0.0,
+        initializer_range=0.02,
+        **super_kwargs,
+    ):
+        super().__init__(**super_kwargs)
+        self.hidden_size = hidden_size
+        self.intermediate_size = intermediate_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.patch_size = patch_size
+        self.image_size = image_size
+        self.initializer_range = initializer_range
+        self.attention_dropout = attention_dropout
+        self.layer_norm_eps = layer_norm_eps
+        self.num_image_positions = num_image_positions
+        self.hidden_act = hidden_act
+
+
+class MolmoPoolingConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MolmoAdapterModel`]. It is used to instantiate an
+    `MolmoAdapterModel` according to the specified arguments, defining the model architecture. Instantiating a configuration
```

Comment:
bit weird to me that MolmoPoolingConfig does not initialize a MolmoPoolingModel and vice versa

Reply:
can rename to MolmoAdapterConfig :smile: was an initial naming decision which never changed afterwards...